src/main/scala/com/gu/contentapi/utils/PredicateUtils.scala

```diff
@@ -0,0 +1,20 @@
+package com.gu.contentapi.utils
+
+object PredicateUtils {
+
+  implicit class RichPredicate[A](val f: Function1[A, Boolean]) extends AnyVal {
+    def apply(v: A) = f(v)
+
+    def &&(g: Function1[A, Boolean]): Function1[A, Boolean] = {
```

Comment:
> why we're not simply using another filter?
Because it is simply not beautiful and more difficult to read than a boolean operator!

>  we're not using 3 of the 4 operators defined
we are using 1 of 3, I am fine removing the 2 others one but this is a generic plumbing that is useful, so usually you add at least `&&` and `||`.

> code feels unnecessary complicated 
How `AcastLog.onlyDownloads && AcastLog.onlySuccessfulReponses` could look complicated? This is consistent with a boolean value operation. There is nothing misleading here, it just behave as you would expect.

> where the && operator is overloaded

The `&& operator` is not overloaded! We define a new method `&&` which does not exist on `Function1[A, Boolean]` type.